src/torchjd/autojac/_utils.py

```diff
@@ -52,7 +52,7 @@ def _get_leaves_of_autograd_graph(roots: list[Tensor], excluded: set[Tensor]) ->
     while nodes_to_traverse:
         current_node = nodes_to_traverse.pop()
 
-        if hasattr(current_node, "variable"):
+        if hasattr(current_node, "variable") and current_node.variable not in excluded:
```

Comment:
This is wrong I think, since then we go to the else case when `current_node.variable in excluded`. We need a nested if.

Reply:
I don't think this can break because when `current_node` has a `variable`, it is I think always an `AccumulateGrad`, and it thus never has any `next_function`. But I agree that this is weird and risky (basically mixing abstraction levels). My next implementation will solve this kind of issue.

Reply:
If you do have in mind an example that would break with this, add a test for it so that we can make sure that the final implementation is not affected.